codesign/archive.go

```diff
@@ -0,0 +1,52 @@
+package codesign
+
+import (
+	"fmt"
+
+	"github.com/bitrise-io/go-xcode/autocodesign"
+	"github.com/bitrise-io/go-xcode/xcarchive"
+)
+
+// Archive ...
+type Archive struct {
+	archive xcarchive.IosArchive
+}
+
+// NewArchive ...
+func NewArchive(archive xcarchive.IosArchive) Archive {
+	return Archive{
+		archive: archive,
+	}
+}
+
+// IsSigningManagedAutomatically ...
+func (a Archive) IsSigningManagedAutomatically() (bool, error) {
+	return a.archive.IsXcodeManaged(), nil
+}
+
+// Platform ...
+func (a Archive) Platform() (autocodesign.Platform, error) {
+	platformName := a.archive.Application.InfoPlist["DTPlatformName"]
+	switch platformName {
+	case "iphoneos":
+		return autocodesign.IOS, nil
+	case "appletvos":
+		return autocodesign.TVOS, nil
+	default:
+		return "", fmt.Errorf("unsupported platform found: %s", platformName)
+	}
+}
+
+// GetAppLayout ...
+func (a Archive) GetAppLayout(uiTestTargets bool) (autocodesign.AppLayout, error) {
+	layout, err := a.archive.ReadCodesignParameters()
+	if err != nil {
+		return autocodesign.AppLayout{}, err
+	}
+	return *layout, nil
+}
+
+// ForceCodesignAssets ...
+func (a Archive) ForceCodesignAssets(distribution autocodesign.DistributionType, codesignAssetsByDistributionType map[autocodesign.DistributionType]autocodesign.AppCodesignAssets) error {
+	return nil
+}
```

Comment:
I have a feeling that we do not have the best abstraction. Maybe it comes from the unfortunate `Project` naming which for me means `xcodeproj` and I would not even think about an archive. Also we cannot force set signing parameters as in a project so (in my mind) we only partially conform to the interface.  

For me something along the lines of `SigningDetailsProvidable` and `SigningDetailsSettable` (sorry for the Swift protocol names) would seem like a better separation than just to have a `Project` interface.

Let me just say that I do not have a better idea on really how I would do it. Everything probably requires a lot of refactoring and I am not sure yet my opinion fully settled. So in overall I am happy to leave it as well and just want to hear your opinion about it.

Reply:
Separated `codesign.DetailsProvider` and `codesign.AssetWriter` in https://github.com/bitrise-io/go-xcode/pull/141/commits/7eb170b841166e5182990b48779f8f51008a2db3